cmd/syncthing/main.go

```diff
@@ -207,132 +207,145 @@ var (
+	flag.StringVar(&options.upgradeTo, "upgrade-to", options.upgradeTo, "Force upgrade directly from specified URL")
+	flag.BoolVar(&options.auditEnabled, "audit", false, "Write events to audit file")
+	flag.BoolVar(&options.verbose, "verbose", false, "Print verbose log output")
+	flag.BoolVar(&options.paused, "paused", false, "Start with all devices paused")
+	flag.StringVar(&options.logFile, "logfile", options.logFile, "Log file name (use \"-\" for stdout)")
+	if runtime.GOOS == "windows" {
+		// Allow user to hide the console window
+		flag.BoolVar(&options.hideConsole, "no-console", false, "Hide console window")
+	}
 
 	longUsage := fmt.Sprintf(extraUsage, baseDirs["config"], debugFacilities())
 	flag.Usage = usageFor(flag.CommandLine, usage, longUsage)
 	flag.Parse()
+
+	return options
 }
 
 func main() {
-	parseCommandLineOptions()
+	options := parseCommandLineOptions()
+	l.SetFlags(options.logFlags)
 
-	if guiAddress != "" {
+	if options.guiAddress != "" {
 		// The config picks this up from the environment.
-		os.Setenv("STGUIADDRESS", guiAddress)
+		os.Setenv("STGUIADDRESS", options.guiAddress)
 	}
-	if guiAPIKey != "" {
+	if options.guiAPIKey != "" {
 		// The config picks this up from the environment.
-		os.Setenv("STGUIAPIKEY", guiAPIKey)
+		os.Setenv("STGUIAPIKEY", options.guiAPIKey)
 	}
 
-	if noConsole {
+	if options.hideConsole {
 		osutil.HideConsole()
 	}
 
-	if confDir != "" {
+	if options.confDir != "" {
 		// Not set as default above because the string can be really long.
-		baseDirs["config"] = confDir
+		baseDirs["config"] = options.confDir
 	}
 
 	if err := expandLocations(); err != nil {
 		l.Fatalln(err)
 	}
 
-	if guiAssets == "" {
-		guiAssets = locations[locGUIAssets]
-	}
-
-	if logFile == "" {
-		// Use the default log file location
-		logFile = locations[locLogFile]
-	}
-
-	if showVersion {
+	if options.showVersion {
 		fmt.Println(LongVersion)
 		return
 	}
 
-	if browserOnly {
+	if options.browserOnly {
 		openGUI()
 		return
 	}
 
-	l.SetFlags(logFlags)
-
-	if generateDir != "" {
-		generate(generateDir)
+	if options.generateDir != "" {
+		generate(options.generateDir)
 		return
 	}
 
 	// Ensure that our home directory exists.
 	ensureDir(baseDirs["config"], 0700)
 
-	if upgradeTo != "" {
-		err := upgrade.ToURL(upgradeTo)
+	if options.upgradeTo != "" {
+		err := upgrade.ToURL(options.upgradeTo)
 		if err != nil {
 			l.Fatalln("Upgrade:", err) // exits 1
 		}
-		l.Okln("Upgraded from", upgradeTo)
+		l.Okln("Upgraded from", options.upgradeTo)
 		return
 	}
 
-	if doUpgradeCheck {
+	if options.doUpgradeCheck {
 		checkUpgrade()
 		return
 	}
 
-	if doUpgrade {
+	if options.doUpgrade {
 		release := checkUpgrade()
 		performUpgrade(release)
 		return
 	}
 
-	if reset {
+	if options.reset {
 		resetDB()
 		return
 	}
 
-	if noRestart {
-		syncthingMain()
+	if options.noRestart {
+		syncthingMain(options)
 	} else {
-		monitorMain()
+		monitorMain(options.logFile)
```

Comment:
less pass the options, as I bet there will be soon something we will want from there.